components/d2l-activity-editor/d2l-activity-content-editor/state/content.js

```
@@ -21,21 +38,80 @@ export class Content {
 		return this;
 	}
 
-	load(entity) {
-		this._entity = entity;
-		this.title = entity.title();
+	async fetchContentModule() {
+		const sirenEntity = await fetchEntity(this._contentModuleHref, this.token);
+		if (sirenEntity) {
+			const entity = new ContentModuleEntity(sirenEntity, this.token, { remove: () => { } });
+			this.loadContentModule(entity);
+		}
+		return this;
+	}
+
+	load(contentEntity) {
+		this._entity = contentEntity;
+		this._contentModuleHref = contentEntity.getModuleHref();
+		this.entityType = contentEntity.getEntityType();
+		if (this._contentModuleHref && this.entityType === CONTENT_TYPES.module) {
+			this.fetchContentModule();
+		}
+	}
+
+	loadContentModule(moduleEntity) {
+		this._contentModule = moduleEntity;
+		this.moduleTitle = moduleEntity.title();
+		this.moduleDescriptionRichText = moduleEntity.descriptionRichText();
+	}
+
+	async save() {
+		// TODO add more save functions as we add more content types
+		if (this.entityType === CONTENT_TYPES.module) {
+			await this.saveContentModule();
+		}
+	}
+
+	async saveContentModule() {
+		if (!this._contentModule) {
+			return;
+		}
+		try {
+			await this._contentModule.setModuleTitle(this.moduleTitle);
+			await this._contentModule.setModuleDescription(this.moduleDescriptionRichText);
+			await this.fetch();
+		} catch (err) {
+			// TODO - what do we want to do with the error?
+		}
```

I know swallowing errors is not recommended but wasn't sure how we want to handle it. Do we want to log the error in telemetry? Do we just want to remove the try/catch and let it blow up? Thoughts?

---

Ideally we would have some kind of error label / toast we could display. I guess nothing like that is currently  in place?

---

A brief investigation leads me to believe nothing like that is currently in place. Although, I don't know how hard it would be to display a toast message if we already have pre-built components for that. 🤔

---

I'm sort of surprised this hasn't come up for assignments. Do they just swallow I guess? Seems bad. Maybe we don't have UX designs for the error cases yet? We might need to reach out to face-dev or Tara to see what should be done ideally.

---

After looking into it a bit more it would look as if by design I'm not supposed to have a try/catch and then let the container handle the thrown error in the container [mixin here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/mixins/d2l-activity-editor-container-mixin.js#L152). That would prevent us from displaying a toast though. Are we okay with that?

---

The even have a comment saying they're not handling errors right now 😓 
```
// TODO - Once we decide how we want to handle errors we may want to add error handling logic to the save
```

---

Yeah, let's do that. It's good to have the error handling centralized like that. Of course, it should do something.... but that's for future work.

---

Sounds good, will remove the try/catch. Agreed that it should handle errors better though but at least it's a known issue.